From 300c02a97728d3c74711cd001da1d294288e6be6 Mon Sep 17 00:00:00 2001
From: Yclept Nemo <orbisvicis@gmail.com>
Date: Sat, 25 Jan 2014 10:28:06 -0500
Subject: [PATCH 1/2] Avoid conflict between shared/static libs in MSVC

    Prepend the "lib" prefix when building static libraries. This only
    affects MSVC, all other platforms already default to the "lib"
    prefix.
    By default import libraries and static libraries share the same name
    in MSVC. This patch avoids name conflicts between shared and static
    libraries built in the same tree using MSVC:
        shared: flandmark.dll
        import: flandmark.lib
        static: libflandmark.lib
---
 libflandmark/CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/libflandmark/CMakeLists.txt b/libflandmark/CMakeLists.txt
index e707f26..ddfa0e1 100644
--- a/libflandmark/CMakeLists.txt
+++ b/libflandmark/CMakeLists.txt
@@ -2,6 +2,9 @@ find_package( OpenCV REQUIRED )
 include_directories(${OpenCV_INCLUDE_DIRS})
 
 add_library(flandmark flandmark_detector.cpp flandmark_detector.h liblbp.cpp liblbp.h)
+if(NOT ${BUILD_SHARED_LIBS})
+  set_target_properties(flandmark PROPERTIES PREFIX "lib")
+endif()
 target_link_libraries(flandmark ${OpenCV_LIBS})
 set_target_properties(flandmark PROPERTIES POSITION_INDEPENDENT_CODE TRUE)
 set_target_properties(flandmark PROPERTIES
-- 
1.8.2.2

